components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/TypeConversionHandler.scala

```diff
@@ -86,6 +96,35 @@ object TypeConversionHandler {
       handleLooseNumberConversion(boxedGivenClass, boxedSuperclassCandidate)
   }
 
+  private def handleIndexedRecordToMapConversion(
+      from: SingleTypingResult,
+      to: TypedClass,
+      strict: Boolean
+  ): Boolean = {
+    if (!AssignabilityUtil.isAssignableToLoadableClass(
+        from.runtimeObjType.klass,
+        "org.apache.avro.generic.IndexedRecord"
+      ) ||
+      !ClassUtils.isAssignable(to.runtimeObjType.klass, classOf[java.util.Map[_, _]])) {
+      false
+    } else {
+      val indexedRecordKeyParam = Typed.genericTypeClass(classOf[String], List())
+      val indexedRecordValueParam = from match {
+        case fromCasted: TypedObjectTypingResult =>
+          superTypeOfTypes(fromCasted.fields.values)
+        case _ => Unknown
+      }
+      val isAssignable =
+        if (strict)
+          AssignabilityDeterminer.isAssignableStrict _
+        else
+          AssignabilityDeterminer.isAssignableLoose _
+
+      isAssignable(indexedRecordKeyParam, to.params.headOption.getOrElse(Unknown)).isValid &&
```

Comment:
1. This `.headOption` / `.drop(1)` is difficult to read, we have pattern matching in scala - take a look at `AssignabilityDeterminer.isSingleAssignableToTypedClass` where we do similar thing
2. For `strict` strategy we shouldn't check it at all. `strict` strategy mean the same as `isAssignable` in java, but witch some tweaks such as unions, unknown etc. In java, `IndexedRecord` can't be converted to `Map`